src/main/java/com/microsoft/aad/adal4j/AuthenticationContext.java

```diff
@@ -104,6 +106,15 @@ public AuthenticationContext(final String authority,
                 this.getAuthority()), this.shouldValidateAuthority());
     }
 
+    public Proxy getProxy() {
+        return proxy;
+    }
+
+    public void setProxy(Proxy proxy) {
+        this.proxy = proxy;
+        authenticationAuthority.setProxy(proxy);

```

Comment:
Not sure about these changes. They look weird. I changed it in order to avoid constructor bloating.